spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/catalog/KubernetesCatalogWatchAutoConfiguration.java

```diff
@@ -39,13 +40,14 @@
  */
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnDiscoveryEnabled
+@ConditionalOnKubernetesCatalogEnabled
 @ConditionalOnCloudPlatform(CloudPlatform.KUBERNETES)
 @AutoConfigureAfter({ KubernetesClientAutoConfiguration.class })
+@EnableConfigurationProperties(KubernetesDiscoveryProperties.class)
```

Comment:
add `@EnableConfigurationProperties(KubernetesDiscoveryProperties.class)` as it was indeed an issue

Reply:
I guess my only question is what happens when we enable configuration properties twice on two different autoconfigurations?  

I wonder if its best to just move the annotation to its own autoconfiguration (thats basically an empty class) because its needed in multiple places.  Thoughts?

Reply:
excellent point, thank you!